### ISSUE TITLE
running own etherpad-light instance: Sorry, this content cannot be embedded

### ISSUE THREAD

**Issue by flyinghuman** [turt2live/matrix-dimension#356]
Hello, i setup dimension to work with our own RIOT and matrix-synapse homeserver. So far so good.
Now we would like to have our own hosted etherpad-light integrated with RIOT. It is on the same server like riot is and can accessed via https://myriothost.tld/pad using nginx reverse proxy.

If we change the defautl url for the widget to our own etherpad-light a error message shows up: Sorry, this content cannot be embedded. What is the reason for that? can anyone here help me? Thanks!
![Etherpad_error](https://user-images.githubusercontent.com/3713548/85142624-f7065c00-b248-11ea-840f-42b0693a7505.JPG)

**Comment by flyinghuman** [comment]
i found a plugin for add embedding headers which solved my problem: https://www.npmjs.com/package/ep_helmet

i only must add my dimension domain name in the etherpad seetings file:

```
"ep_helmet":{
  "frameguard":{
    "action": "allow-from",
    "domain": "https://DOMAINIFRAMEWILLBEHOSTEDIN.com"
  }
}
```

**Comment by flyinghuman** [comment]
nope, today the same error occured. reopening this issue. what can i do? configure nginx with special options? thanks!

**Comment by TheOneWithTheBraid** [comment]
@flyinghuman Did you somehow fix the issue? I encounter the same issue.

**Comment by rumsant** [comment]
I am having this same problem.

**Comment by dimm0** [comment]
Same here

**Comment by flyinghuman** [comment]
> @flyinghuman Did you somehow fix the issue? I encounter the same issue.

yes, it is working for me with nginx settings:

`location /pad {
                rewrite                /pad/(.*) /$1 break;
                rewrite                ^/pad$ /pad/ permanent;
                proxy_pass             http://localhost:9001/;
                proxy_pass_header Server;
                proxy_redirect         / /pad/;
                proxy_set_header       Host $host;
                proxy_buffering off;
                proxy_hide_header X-Frame-Options;
        }`

**Comment by dimm0** [comment]
I fixed the error above by removing the room name from URL. It didn't like the "!" symbol...
Now I'm having CORS issues :(

**Comment by flyinghuman** [comment]
> I fixed the error above by removing the room name from URL. It didn't like the "!" symbol...
> Now I'm having CORS issues :(

try the helmet plugin in etherpad-lite and the above mentioned nginx `proxy_hide_header X-Frame-Options;`

**Comment by dimm0** [comment]
Ok, I fixed it without the plugin. Simply removing the $room makes the trick, since private rooms have "!" in the name
